pkg/diagnostics/http_tracing.go

```diff
@@ -109,9 +90,11 @@ func SpanContextFromRequest(req *fasthttp.Request) (sc trace.SpanContext, ok boo
 
 // SpanContextToRequest modifies the given request to include traceparent and tracestate headers.
 func SpanContextToRequest(sc trace.SpanContext, req *fasthttp.Request) {
-	h := SpanContextToString(sc)
-	req.Header.Set(traceparentHeader, h)
-	tracestateToRequest(sc, req)
+	if (sc != trace.SpanContext{}) {
```

Comment:
Same as above, under which condition is this expected to be `!=`?

Reply:
correct. sc will be empty object and no ops if trace is disabled.

Reply:
You probably need parentheses here for the equality check:

https://stackoverflow.com/questions/28447297/how-to-check-for-an-empty-struct

Reply:
fixed and added unit-tests